server/etcdserver/api/snap/snapshotter.go

```diff
@@ -129,83 +129,68 @@ func (s *Snapshotter) loadMatching(matchFn func(*raftpb.Snapshot) bool) (*raftpb
 	}
 	var snap *raftpb.Snapshot
 	for _, name := range names {
-		if snap, err = loadSnap(s.lg, s.dir, name); err == nil && matchFn(snap) {
+		if snap, err = s.loadSnap(s.dir, name); err == nil && matchFn(snap) {
 			return snap, nil
 		}
 	}
 	return nil, ErrNoSnapshot
 }
 
-func loadSnap(lg *zap.Logger, dir, name string) (*raftpb.Snapshot, error) {
+func (s *Snapshotter) loadSnap(dir, name string) (*raftpb.Snapshot, error) {
 	fpath := filepath.Join(dir, name)
-	snap, err := Read(lg, fpath)
+	snap, err := Read(s.lg, fpath)
 	if err != nil {
 		brokenPath := fpath + ".broken"
-		if lg != nil {
-			lg.Warn("failed to read a snap file", zap.String("path", fpath), zap.Error(err))
-		}
+		s.lg.Warn("failed to read a snap file", zap.String("path", fpath), zap.Error(err))
 		if rerr := os.Rename(fpath, brokenPath); rerr != nil {
-			if lg != nil {
-				lg.Warn("failed to rename a broken snap file", zap.String("path", fpath), zap.String("broken-path", brokenPath), zap.Error(rerr))
-			}
+			s.lg.Warn("failed to rename a broken snap file", zap.String("path", fpath), zap.String("broken-path", brokenPath), zap.Error(rerr))
 		} else {
-			if lg != nil {
-				lg.Warn("renamed to a broken snap file", zap.String("path", fpath), zap.String("broken-path", brokenPath))
-			}
+			s.lg.Warn("renamed to a broken snap file", zap.String("path", fpath), zap.String("broken-path", brokenPath))
 		}
 	}
 	return snap, err
 }
 
 // Read reads the snapshot named by snapname and returns the snapshot.
 func Read(lg *zap.Logger, snapname string) (*raftpb.Snapshot, error) {
+	if lg == nil {
+		lg = zap.L()
+	}
```

Comment:
```suggestion
verify.Assert(lg != nil, "the logger should not be nil")
```

You need to add import `"go.etcd.io/etcd/client/pkg/v3/verify"`

Reply:
Thanks. Have updated it.